src/ucp/api/ucp.h

```diff
@@ -631,6 +633,120 @@ void ucp_worker_progress(ucp_worker_h worker);
+ *
+ * There are two alternative ways to use the wakeup mechanism: the first is the
+ * file descriptor obtained per worker (this function) and the second is the
+ * @ref ucp_worker_wait function for waiting on the next event internally.
+ *
+ * @param [in]  worker    Worker of notified events.
+ * @param [out] fd        File descriptor.
+ */
+ucs_status_t ucp_worker_get_efd(ucp_worker_h worker, int *fd_p);
+
+
+/**
+ * @ingroup UCP_WAKEUP
+ * @brief Wait for an event of the worker.
+ *
+ * This routine waits (blocking) until an event has happened, as part of the
+ * wake-up mechanism.
+ *
+ * The worker must be armed before waiting on an event (must be re-armed after
+ * it has been signaled for re-use) with @ref ucp_worker_arm.
+ * The events triggering the return of this function depend on the
+ * interfaces used by the worker and are defined in the transport layer, and
+ * typically represent a request completion or newly available resources.
+ * It can also be triggered by calling @ref ucp_worker_signal .
+ *
+ * The wake-up mechanism exists to allow for the user process to register for
+ * notifications on events of the underlying interfaces, and wait until such
+ * occur. This is an alternative to repeated polling for request completion.
+ * The goal is to allow for waiting while consuming minimal resources from the
+ * system. This is recommended for cases where traffic is infrequent, and
+ * latency can be traded for lower resource consumption while waiting for it.
+ *
+ * There are two alternative ways to use the wakeup mechanism: the first is the
+ * file descriptor obtained per worker using @ref ucp_worker_get_efd and the
+ * second is waiting on the next event internally (this function).
+ *
+ * @note During the blocking call the wake-up mechanism relies on other means of
+ * notification and may not progress some of the requests as it would when
+ * calling @ref ucp_worker_progress (which is not invoked in that duration).
+ *
+ * @param [in]  worker    Worker to wait for events on.
+ */
+ucs_status_t ucp_worker_wait(ucp_worker_h worker);
+
+
+/**
+ * @ingroup UCP_WAKEUP
+ * @brief Turn on event notification for the next event.
+ *
+ * This routine needs to be called before waiting on each notification on this
+ * worker, so will typically be called once the processing of the previous event
+ * is over, as part of the wake-up mechanism.
+ *
+ * The worker must be armed before waiting on an event (must be re-armed after
+ * it has been signaled for re-use) with @ref ucp_worker_arm.
+ * The events triggering a signal of the file descriptor from
+ * @ref ucp_worker_get_efd , or the return of a call to @ref ucp_worker_wait -
+ * depend on the interfaces used by the worker and are defined in the transport
+ * layer, and typically represent a request completion or newly available
+ * resources. It can also be triggered by calling @ref ucp_worker_signal .
+ *
+ * @example
```

Comment:
Did you try to generate doxygen ? it will not work.
The code has to be protected with @code / @endcode